src/sliceList.ml

```diff
@@ -143,7 +145,8 @@ let[@inline] slice_list_remove (l:slice_list) (auctions:liquidation_auctions) (e
   assert (meta.slice_list_burrow = slice.contents.burrow);
   match meta.slice_list_bounds with
   (* FIXME: Perhaps throw specific error code here? *)
-  | None -> (failwith "Attempting to remove an element from an empty list" : liquidation_auctions*slice_list*avl_ptr*liquidation_slice_contents)
+  (* FIXME: George: It is specific now, but I still think this is internal, right? *)
```

Comment:
It is not very easy tell, but I feel like it should be internal. The most likely place it might be exposed is `touch_liquidation_slice` endpoint, but we call `ensure_valid_leaf_ptr` initially, so we know that it corresponds to a leaf. And once there is a leaf, we know that its burrow has a non-empty slice list.

But it's not "obviously correct", so it is concerning. Maybe we should distinguish the `leaf_ptr`'s we get from user which might be incorrect and `leaf_ptr`'s that we know to be correct. (eg. having `unsafe_leaf_ptr`, `leaf_ptr` and `ensure_valid_leaf_ptr: storage -> unsafe_leaf_ptr -> leaf_ptr`). But not in scope of this PR, probably :).

Reply:
Good point! I also think it's internal. I'm gonna leave a `FIXME` comment here so that we revisit this later and make sure it is internal in all cases :+1: